cassandra/murmur3.py

```diff
@@ -0,0 +1,110 @@
+from six.moves import range
+import struct
+
+
+def body_and_tail(data):
+    l = len(data)
+    nblocks = l // 16
+    tail = l % 16
+    if nblocks:
+        return struct.unpack_from('qq' * nblocks, data), struct.unpack_from('b' * tail, data, -tail), l
+    else:
+        return tuple(), struct.unpack_from('b' * tail, data, -tail), l
+
+
+def rotl64(x, r):
+    mask = 2 ** r - 1
+    rotated = (x << r) | ((x >> 64 - r) & mask)
```

Comment:
The left shift doesn't mask off bits that extend beyond 64 bits. Example:

>>> hex(rotl64(0x8000000000000000, 1))
'0x10000000000000001L'
should be: 
'0x1L'

Does this present a problem? Or does the final truncation take care of this?